src/custom/state/claim/hooks/index.ts

```diff
@@ -652,3 +668,39 @@ export function useClaimDispatchers() {
 export function useClaimState() {
   return useSelector((state: AppState) => state.claim)
 }
+
+/**
+ * Gets an array of available claims parsed and sorted for the UI
+ *
+ * Syntactic sugar on top of `useUserClaims`
+ *
+ * @param account
+ */
+export function useUserAvailableParsedClaims(account: Account): UserClaimDataDetails[] {
+  const { available } = useClassifiedUserClaims(account)
+  const { chainId } = useActiveWeb3React()
+
+  const checkType = useCallback((type) => Number(FREE_CLAIM_TYPES.includes(type)), [])
+
+  const sorted = useMemo(() => available.sort((a, b) => checkType(b.type) - checkType(a.type)), [available, checkType])
+
+  const parsed = useMemo(() => {
+    return sorted.map((claim: UserClaimData) => {
+      const currencyAmount = parseClaimAmount(claim.amount, chainId)
+      const price = mapTypeToPrice(claim.type, chainId)
+      const currency = mapTypeToCurrency(claim.type, chainId)
+      const cost = price && currencyAmount && Number(formatSmart(price)) * Number(formatSmart(currencyAmount))
```

Comment:
Not sure if the is the best way to multiply these 2 values

Reply:
Both price and currencyAmount are CurrencyAmount's and they have correct values.

Reply:
For example, when called formatSmart on price and currencyAmount we get 
```
price 0.000038
currencyAmount 4754
```
so those 2 multiplied should give `0.180652`

Reply:
Not at all 😅

We will never multiply `number` for calculations. `number` operation has floating point rouding errors.

Also, it's not good to do `Number(formatSmart(price))`, its dirty and also will fail depending on the locale of the format.

CurrencyAmount has already operations: 
![image](https://user-images.githubusercontent.com/2352112/149356737-7491ea7d-b496-4ac6-8c70-17a787031147.png)

Reply:
Then, once u get another resulting `CurrencyAmount` Then maybe u can `formatSmart` it?

Reply:
Updated this!